### DISCUSSION TITLE
Upon registration, Click on a link to activate account, instead copy & fill confirmation code from email

### DISCUSSION THREAD

**Discussion by bebekmacho69** [codeigniter4/shield#645] [Q&A]
Hello.

So I want for user to activate their account by simply clicking on a link from their received email, I've tried to copy the form inside email activation page into the email html template like the following : 
```html
<form action="http://localhost:8080/index.php/auth/a/verify" method="post">
    <input type="hidden" name="csrf_test_name" value="bc674b86bd56988fb30c02d0e28cb05d">
    <input type="hidden" name="token" value="762341">
    <button type="submit">Activate your account</button>
</form>
```
but it wont work, showing 404 page when clicked, reading on various sites said that most email clients blocks html form
is there any way to overcome this? thank you...

**Comment by datamweb** [comment]
Hi,
Sending such a form to email is not common, and definitely the email server will give warnings to the user. If you want the user to be activated by clicking on the link, you can create a personal **EmailActivator** class that supports activation with the link as a get method.

http://localhost:8080/index.php/auth/my/verify?token=123456

More information refer to [defining-new-actions](https://github.com/codeigniter4/shield/blob/develop/docs/auth_actions.md#defining-new-actions).

**Reply by bebekmacho69** [reply]
Okey , so custom EmailActivator is needed..
Will try, and I will share if done in case someone want registeration flow to be the same.
Thank you for the answer..
